modules/internal/Bytes.chpl

```diff
@@ -83,6 +83,166 @@ module Bytes {
     }
   }
 
+  //
+  // createBytes* functions
+  //
+
+  /*
+    Creates a new bytes which borrows the internal buffer of another bytes. If
+    the buffer is freed before the bytes returned from this function, accessing
+    it is undefined behavior.
+
+    :arg s: Object to borrow the buffer from
+    :type s: `bytes`
+
+    :returns: A new `bytes`
+  */
+  inline proc createBytesWithBorrowedBuffer(s: bytes) {
```

Comment:
I wonder if these would be more reasonable in the documentation if we had a single generic version that had compilerErrors if the argument type wasn't applicable? Then we'd just document that one and indicate the types it accepts in the `:arg s:` chpldoc.

Oh, maybe they need to be separate, because some have more additional arguments than others... As is is OK.